superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -549,7 +555,10 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
         DB.backend === db?.engine || DB.engine === db?.engine,
     ) as DatabaseObject
   )?.engine_information?.allow_ssh_tunneling;
-  const sshTunneling = isFeatureEnabled(FeatureFlag.SSH_TUNNELING);
+  const sshTunneling =
+    isFeatureEnabled(FeatureFlag.SSH_TUNNELING) &&
+    engineAllowsSSHTunneling !== undefined &&
+    engineAllowsSSHTunneling;
```

Comment:
```suggestion
    isFeatureEnabled(FeatureFlag.SSH_TUNNELING) &&
    engineAllowsSSHTunneling;
```
Could this be simplified if just checking for truthiness? Or is there something specific that needs the `!== undefined` check?

Reply:
i had to explicitly add undefined for typing issues to go away since `engineAllowsSSHTunneling` key isn't always set for every engine coming back from the `available` endpoint

Reply:
```suggestion
    isFeatureEnabled(FeatureFlag.SSH_TUNNELING) &&
    engineAllowsSSHTunneling !== undefined;
```
That makes sense. In that case, would `engineAllowsSSHTunneling` be needed at the end? I think `engineAllowsSSHTunneling !== undefined` covers what it is checking for.